### ISSUE TITLE
[BITV] 9.1.4.11/6.3 - All buttons "Change scope level of ..." do not meet the contrast requirements for graphical user interface components - actual contrast is 2.8:1. (1)

### ISSUE THREAD

**Issue by AndyScherzinger** [nextcloud/server#36975]
- [ ] Fix
- [ ] Backport

![97384da12ad39b1f7335e094bda19bb2](https://user-images.githubusercontent.com/1315170/222243578-7ee076c4-ec65-4d9e-a664-6c201adb73d0.png)

<details>
<summary>Details</summary>

https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-3dbfb7615a-v6-n3

</details>

**Comment by JuliaKirschenheuter** [comment]
Dear @jancborchardt, @nimishavijay, @marcoambrosini 

Here we have a next accessibility problem. The first thought how to fix this issue was to increase a contrast for interactive element. But the problem is that this icon-button is actually a dropdown. And dropdown have to be a dropdown and not a icon-button.

![Screenshot from 2023-06-22 11-06-30](https://github.com/nextcloud/server/assets/6078378/c58d44de-2c70-48d1-aee0-3814740fe53b)

- Should i replace all this buttons with NcSelect? But i think that would look like "too much in one place". 
- And besides this we have to restructure "sections" (full name, email, phone number ...) because right now we have an input fields which are connected to there labels. Where should a dropdown (NcSelect) belong to?

Please give me a feedback. Thanks a lot!

**Comment by JuliaKirschenheuter** [comment]
Probably action buttons are ok. I will check them again.

**Comment by nimishavijay** [comment]
Agreed that using a dropdown component in place of the icon wouldn't be the best solution. 

Possibly this solution is out of scope, but during the initial designs for Nextcloud profile, there were some ideas for a single column layout for the settings, which would look similar to the new contacts layout. This would allow more space for using a dropdown component if required. What do you think about this?

**Comment by JuliaKirschenheuter** [comment]
Thanks @nimishavijay for your answer!
Do you mean that we can use dropdowns (select) instead and after some time would have a new design look for this page that everything would look much nicer with more place?

**Comment by nimishavijay** [comment]
@JuliaKirschenheuter With the current layout we cannot use dropdown components, because as you mentioned it will look "too much".
If we switch to a different layout, then we will have enough space to use dropdowns, and everything will look nicer.
If a layout change is too much work for this issue, then we won't use dropdowns and we can find another solution :) What do you think? I will share mockups of the new layout or look for a different solution depending on what's more suitable for you :)

**Comment by JuliaKirschenheuter** [comment]
@nimishavijay could we probably just use an action button just like everywhere?

![Screenshot from 2023-07-11 17-23-09](https://github.com/nextcloud/server/assets/6078378/6518b63f-53ef-4cdb-b076-55c217190c5b)

**Comment by jancborchardt** [comment]
> Could we probably just use an action button like everywhere?

@JuliaKirschenheuter yes that would be the simplest solution and focus on fixing the reported issue – let’s go with that. :)

**Comment by AndyScherzinger** [comment]
Can we check this one with @michaelnissenbaum I would hope that it being a button component and the icon having the right contrast to be "enough". I really wouldn't want to have all these items to have a borders by default, but am happy to be proven otherwise.

**Comment by JuliaKirschenheuter** [comment]
We had an information that this action buttons were not enough discoverable even if they would have higher contrast. Or would it be really enough if we just increase a contrast on icon-only **action button**?  
@michaelnissenbaum thanks for your help!

**Comment by michaelnissenbaum** [comment]
Hi @JuliaKirschenheuter. All interactive graphical controls implemented using icons, SVGs, or elements like "img" need to have a minimum contrast ratio of 3:1 with the background. They should also be clearly distinguishable without keyboard focus or mouseover interactions.

**Comment by AndyScherzinger** [comment]
@michaelnissenbaum for me your statement than translates to the icon needs to have the right contrast _but_ there is no need to have a border around it (in an unselected/-focussed state)

**Comment by michaelnissenbaum** [comment]
@AndyScherzinger That's correct.

**Comment by AndyScherzinger** [comment]
@nextcloud/designers fine by you if we then just raise the contrast on the svgs? I would expect yes.

**Comment by szaimen** [comment]
From my side yes but it should probably also become a button if not already done. Wasnt there alrready a PR that fixes this somewhere?

**Comment by JuliaKirschenheuter** [comment]
> From my side yes but it should probably also become a button if not already done. Wasnt there alrready a PR that fixes this somewhere?

Not yet, i will push my changes now

**Comment by JuliaKirschenheuter** [comment]
**1. As follow-up: focused (i will open a ticket for it)**
If an action menu is opened, action button itself have to be focused to. For now it is not a case.

![Peek 2023-07-19 14-27](https://github.com/nextcloud/server/assets/6078378/ac36440d-25e4-4158-8f2e-49c31c048ebe)

**2. As follow-up: focused (i will open a ticket for it)** not sure about styles inside opened action menu: should we set a double outline on active + focused / hovered action menu item? @marcoambrosini, @nimishavijay 

![Peek 2023-07-19 14-33](https://github.com/nextcloud/server/assets/6078378/1c3cd6ba-ac7b-4a42-b77c-56ccf3e27f3e)

**Comment by marcoambrosini** [comment]
@JuliaKirschenheuter could you point me to the aria documentation that addresses this? I couldn't find this rule.

As a rule of thumb, we want to avoid displaying those double outlines to people that use mouses, trackpads and fingers as pointing devices. As they already know which part of the page they're interacting with.
